src/pudl/output/pudltabl.py

```diff
@@ -199,141 +208,50 @@ def _get_table_from_db(self, table_name: str, resource: Resource) -> pd.DataFram
+                for df in pd.read_sql(
+                    self._select_between_dates(table_name),
+                    self.pudl_engine,
+                    chunksize=100_000,
+                )
             ]
         )
 
-    def pu_eia860(self, update=False):
-        """Pull a dataframe of EIA plant-utility associations.
+    def _select_between_dates(self, table: str) -> sa.sql.expression.Select:
+        """For a given table, returns an SQL query that filters by date, if specified.
 
-        Args:
-            update (bool): If true, re-calculate the output dataframe, even if
-                a cached version exists.
+        Method uses the PudlTabl ``start_date`` and ``end_date`` attributes.  For EIA
+        and most other tables, it compares ``report_date`` column against start and end
+        dates.  For FERC1 ``report_year`` is used.  If neither ``report_date`` nor
+        ``report_year`` are present, no date filtering is done.
 
-        Returns:
-            pandas.DataFrame: a denormalized table for interactive use.
-        """
-        if update or self._dfs["pu_eia"] is None:
-            self._dfs["pu_eia"] = pudl.output.eia860.plants_utils_eia860(
-                self.pudl_engine, start_date=self.start_date, end_date=self.end_date
-            )
-        return self._dfs["pu_eia"]
-
-    def pu_ferc1(self) -> pd.DataFrame:
-        """Pull a dataframe of FERC plant-utility associations.
+        Arguments:
+            table: name of table to be called in SQL query.
 
         Returns:
-            A denormalized table for interactive use.
-        """
-        return pd.read_sql_table(
-            "denorm_plants_utilities_ferc1", self.pudl_engine
-        ).pipe(apply_pudl_dtypes, group="ferc1")
+            A SQLAlchemy select object restricting the date column (either
+            ``report_date`` or ``report_year``) to lie between ``self.start_date`` and
+            ``self.end_date`` (inclusive).
+        """
+        pt = pudl.output.pudltabl.get_table_meta(self.pudl_engine)
+        tbl = pt[f"{table}"]
+        if "report_date" in tbl.columns:
+            date_col = tbl.c.report_date
+        elif "report_year" in tbl.columns:
+            date_col = tbl.c.report_year
+        else:
+            date_col = None
+        tbl_select = sa.sql.select(tbl)
+        if self.start_date and date_col is not None:
+            start_date = pd.to_datetime(self.start_date)
+            tbl_select = tbl_select.where(date_col >= start_date)
+        if self.end_date and date_col is not None:
+            end_date = pd.to_datetime(self.end_date)
+            tbl_select = tbl_select.where(date_col <= end_date)
+        return tbl_select
 
     ###########################################################################
     # EIA 860/923 OUTPUTS
     ###########################################################################
-    def utils_eia860(self, update=False) -> pd.DataFrame:
```

Comment:
I've been removing the `PudlTabl` methods one by one as the table is replicated, and the new version is verified to be functionally equivalent to the old version.